[Java] Add release notes for 3.53.0 (#464)
SAP/cloud-sdk: docs/java/release-notes.mdx

  If your application relies on the version provided by Cloud SDK, please consider declaring the dependency version yourself.
  The latest version can be found in [Maven Central](https://search.maven.org/artifact/org.jsoup/jsoup).

## New Functionality

- We added convenience methods for `null` comparison in our _OData_ API to avoid unnecessary null-casting:
  - `Entity.FIELD_NAME.equalToNull()` - _OData V4 VDM_ / _Generic OData Client_
  - `Entity.FIELD_NAME.notEqualToNull()` - _OData V4 VDM_ / _Generic OData Client_
  - `Entity.FIELD_NAME.eqNull()` - _OData V2 VDM_
  - `Entity.FIELD_NAME.neNull()` - _OData V2 VDM_

- We now support function import calls in OData V2 using the HTTP GET method inside batch requests. Within the batch request, the function import call can be added to the `addReadOperations()` method. 
You can execute a batch request containing a function import call as follows:
```java
singleValueReturnFunction = service.callSomeFunction();
service.batch()
  .addReadOperations(singleValueReturnFunction)
  .executeRequest(destination);
```
You can read the response of the function by:
```java

> functionResult = batchResponse.getReadResult(singleValueReturnFunction);

Author: Yes, that is documented indeed. I'll link it here.

```

> You can also add functions that return a collection inside batch requests and parse the results in similar manner.

Reviewer: ```suggestion
You can also add functions that return a collection inside batch requests and parse the results yourself in a similar manner.
```

Author: ```suggestion
You can also add functions that return a collection inside batch requests and parse the results yourself in a similar manner.
Please refer to [our documentation](https://sap.github.io/cloud-sdk/docs/java/features/odata/use-typed-odata-v2-client-in-sap-cloud-sdk-for-java#execute-batch-request) for more details.
```

> - (Beta) Reading destinations from environment variables now allows you to configure authorization token forwarding to the target system.

Reviewer: ```suggestion
- [**Beta**] Reading destinations from environment variables now allows you to configure authorization token forwarding to the target system.
```

>   This can be activated in the following ways

Reviewer: ```suggestion
  You can activate in the following ways
```

Author: ```suggestion
  You can activate this feature in the following ways
```

  1. Setting the `authentication` property directly:
      ```
      [
        {
          "type": "HTTP",
          "name": "destinationName",
          "url": "https://URL",
          "authentication": "TokenForwarding"
        }
      ]
      ```
  2. Adding the property `forwardAuthToken=true`, as already known from the approuter:
      ```
      [
        {
          "type": "HTTP",
          "name": "destinationName",
          "url": "https://URL",
          "forwardAuthToken": true
        }
      ]
      ```
  Both of those settings will cause the destination to take the value of the `Authorization` HTTP header contained in the current HTTP request for every request to the specified target system.
  

>   **Note:** This means that token set via the `AuthTokenAccessor.executeWith(...)` will **not** be forwarded to the target system. 

Reviewer: ```suggestion
  **Note:** The settings above will override a token set with `AuthTokenAccessor.executeWith(...)` and it will **not** be passed to the target system. 
```

Author: ```suggestion
  **Note:** The settings above will override a token set with `AuthTokenAccessor.executeWith(...)`. Moreover, authentication tokens set via `AuthTokenAccessor.executeWith(...)` will **not** be forwarded to the target system. 
```

> - We now support adding headers to a specific destination using properties, so they will always be included when calling the target system. 

Reviewer: ```suggestion
- You can now add `headers` to a `destination` using the `property` method so that they will always be included when calling the target system. 
```

>   Usage: `.property("URL.headers.foo1", "bar1")`. Relevant section in documentation: [SDK Docs](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/sdk-connectivity-destination-service#provide-headers-for-destinations).

Reviewer: ```suggestion
  Here's an example of the `property` method call: `.property("URL.headers.foo1", "bar1")`. Find more details on setting properties [here](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/sdk-connectivity-destination-service#provide-headers-for-destinations).
```

> - We now support adding query parameters to a specific destination using properties.

Reviewer: ```suggestion
- You can now add query parameters to a destination using properties.
```

  They will always be included in the query string when calling the target system.

>   The query values should be provided without encoding. They will be encoded in the query string.

Reviewer: ```suggestion
  The query values should be provided without encoding. They will be encoded in the query string by the SDK.
```

>   Example usage: `.property("URL.queries.airport", "Köln Bonn")`. The query string will include `airport=K%C3%B6ln%20Bonn`. 

Reviewer: ```suggestion
  Example of adding a query parameter using `property` method: `.property("URL.queries.airport", "Köln Bonn")`. 
  The rendered query string will look like: `airport=K%C3%B6ln%20Bonn`. 
```

  Relevant section in documentation: [SDK Docs](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/sdk-connectivity-destination-service#add-query-parameters-via-destination-properties)
  
## Improvements

- `HttpClientFactory` is now a `@FunctionalInterface`. With that, (ad-hoc) instances can conveniently be constructed using simply a lambda expression like so: `HttpClientFactory factory = destination -> HttpClients.createDefault();`.

> - Removed unnecessary warning messages when not having an XSUAA or IAS service bound to the application.

Reviewer: ```suggestion
- We removed unnecessary warning messages when generated when an XSUAA or IAS service is not bound to an application.
```

- Dependency Updates:
  - SAP dependency updates:
    - Update BTP Neo Tomcat 8 SDK from `4.6.5` to `4.7.9`
    - Update [XSUAA Token Client and Token Flow API](https://github.com/SAP/cloud-security-xsuaa-integration/tree/master/token-client) from `2.10.3` to `2.10.5`
    - Update Neo JavaEE 7 SDK from `1.102.5` to `1.103.10`
  - Other dependency updates:
    - Minor version updates:
      - Update Spring from `5.3.8` to `5.3.9`
      - Update GSON from `2.8.7` to `2.8.8`
      - Update Togglz from `2.9.8` to `2.9.9`
      - Update Mockito from `3.11.2` to `3.12.3`
      - Update Spring Security from `5.5.1` to `5.5.2`

## Fixed Issues

> - We fixed an issue where HTTP header keys of an OData response were not handled case-insensitive, causing problems with 
>   processing those responses. This is fixed now and should handle headers the same way, regardless of capitalization.

Reviewer: ```suggestion
- We fixed the issue where HTTP header keys of an OData response were not handled case-insensitive, causing problems with processing those responses. 
This is fixed now and should handle headers the same way, regardless of capitalization.
```
